DFC.ServiceTaxonomy.GraphSync/GraphSyncers/Helpers/DescribeContentItemHelper.cs

```diff
@@ -0,0 +1,208 @@
+                            child.RelationshipPathString = relationshipString;
+                        }
+                        else
+                        {
+                            context.CurrentDepth = 1;
+                            child.RelationshipPathString = $@"match (s:{string.Join(":", context.SourceNodeLabels)} {{{context.SourceNodeIdPropertyName}: '{context.SourceNodeId}'}})-[r{1}:{child.Relationship}]-(d{1}:{string.Join(":", child.Destination!)})";
+                        }
+                    }
+                }
+
+                currentList.AddRange(context.AvailableRelationships);
+            }
+
+            foreach (var childContext in context.ChildContexts)
+            {
+                await GetRelationships((IDescribeRelationshipsContext)childContext, currentList, context);
+            }
+
+            return currentList;
+        }
+
+        public async Task BuildRelationships(ContentItem contentItem, IDescribeRelationshipsContext context, string sourceNodeIdPropertyName, string sourceNodeId, IEnumerable<string> sourceNodeLabels)
+        {
+            var contentTypeDefinition = _contentDefinitionManager.GetTypeDefinition(contentItem.ContentType);
+
+            _graphSyncHelper.ContentType = contentItem.ContentType;
+
+            var itemContext = contentItem == RootContentItem ? context : new DescribeRelationshipsContext(sourceNodeIdPropertyName, sourceNodeId, sourceNodeLabels, contentItem, _graphSyncHelper, _contentManager, _publishedContentItemVersion, context, _serviceProvider);
+
+            foreach (var part in contentTypeDefinition.Parts)
+            {
+                var partSyncer = _contentPartGraphSyncers.FirstOrDefault(x => x.PartName == part.Name);
+
+                if (partSyncer != null)
+                {
+                    await partSyncer.AddRelationship(itemContext);
+                }
+
+                foreach (var relationshipField in part.PartDefinition.Fields)
+                {
+                    itemContext.SetContentPartFieldDefinition(relationshipField);
+                    itemContext.SetContentField((JObject)contentItem.Content[relationshipField.PartDefinition.Name][relationshipField.Name]);
+
+                    var fieldSyncer = _contentFieldsGraphSyncers.FirstOrDefault(x => x.FieldTypeName == relationshipField.FieldDefinition.Name);
+
+                    var contentItemIds =
+                           (JArray)contentItem.Content[relationshipField.PartDefinition.Name][relationshipField.Name]
+                               .ContentItemIds;
+
+                    if (contentItemIds != null)
+                    {
+                        foreach (var relatedContentItemId in contentItemIds)
+                        {
+                            await BuildRelationships(await _contentManager.GetAsync(relatedContentItemId.ToString()), itemContext, sourceNodeIdPropertyName, sourceNodeId, sourceNodeLabels);
```

Comment:
This needs to be aware of whether the published or preview graph is being visualised and either the appropriate flag passed to `GetAsync` or there may be an appropriate helper on `ContentItemVersion`.
I think you need to test visualising both the published and preview graphs with different data, to shake out any assumptions elsewhere.